sqlobject/src/test/java/org/jdbi/v3/docs/TestInClauseExpansion.java

```diff
@@ -63,9 +65,24 @@ public void testInClauseExpansion() throws Exception
     }
 
     @UseStringTemplate3StatementLocator
+    @RegisterContainerMapper(ImmutableSetCollectorFactory.class)
     public interface DAO
     {
         @SqlQuery
-        Set<String> findIdsForNames(@BindIn("names") List<Integer> names);
+        ImmutableSet<String> findIdsForNames(@BindIn("names") List<Integer> names);
+    }
+
+    public static class ImmutableSetCollectorFactory<T>
+            implements CollectorFactory<T, ImmutableSet.Builder<T>, ImmutableSet<T>> {
+
+        public boolean accepts(Class<?> type) {
+            return ImmutableSet.class.isAssignableFrom(type);
+        }
+
+        @Override
+        public Collector<T, ImmutableSet.Builder<T>, ImmutableSet<T>> newCollector(Class<ImmutableSet<T>> type) {
+            return Collector.of(ImmutableSet.Builder::new, ImmutableSet.Builder::add, (first, second) -> first,
```

Comment:
With that said, it is so cool that this works with just a few method references.